MatrixSDK/MXSession.m

```diff
@@ -1416,6 +1441,11 @@ - (void)serverSyncWithServerTimeout:(NSUInteger)serverTimeout
                       clientTimeout:(NSUInteger)clientTimeout
                         setPresence:(NSString*)setPresence
 {
+    if (MXSDKOptions.sharedInstance.enableSyncProgress)
+    {
+        [self.syncProgress incrementSyncAttempt];
```

Comment:
You can probably just call this directly without checking, here and in other places in this file. If it's `nil` nothing happens.

Reply:
This is true, though I have burned too many fingers on the implicit obj-c nil behaviour and prefer to be explicit where possible (e.g. incorrect typecast not actually being nil, dictionaries crashing on nil etc)

Reply:
Fair enough 👍